src/widgets/barchart/bar_group.rs

```diff
@@ -21,17 +21,44 @@ pub struct BarGroup<'a> {
 }
 
 impl<'a> BarGroup<'a> {
+    /// Creates a new [`BarGroup`] widget with the given bars and no label.
+    ///
+    /// The `bars` parameter accepts an iterator containing items of type [`Bar`].
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// # use ratatui::{prelude::*, widgets::*};
+    /// BarGroup::new([Bar::new("label", 10), Bar::new("label", 10)]);
+    /// ```
+    pub fn new<I>(bars: I) -> Self
+    where
+        I: IntoIterator<Item = Bar<'a>>,
+    {
+        Self {
+            bars: bars.into_iter().collect(),
+            ..Default::default()
+        }
+    }
+
     /// Set the group label
     #[must_use = "method moves the value of self and returns the modified value"]
-    pub fn label(mut self, label: Line<'a>) -> BarGroup<'a> {
-        self.label = Some(label);
+    pub fn label<T>(mut self, label: T) -> BarGroup<'a>
+    where
+        T: Into<Line<'a>>,
+    {
+        self.label = Some(label.into());
         self
     }
 
     /// Set the bars of the group to be shown
     #[must_use = "method moves the value of self and returns the modified value"]
-    pub fn bars(mut self, bars: &[Bar<'a>]) -> BarGroup<'a> {
-        self.bars = bars.to_vec();
+    pub fn bars<I, T>(mut self, bars: I) -> BarGroup<'a>
```

Comment:
Add a note about conversion (applies generally to the new methods, won't repeat, but do a quick check for this)

Reply:
Probably irrelevant with #995